src/v/cluster/cluster_utils.cc

```diff
@@ -468,6 +468,71 @@ std::optional<ss::sstring> check_result_configuration(
     }
     return {};
 }
+namespace {
+const std::vector<ss::sstring>& stm_snapshot_names() {
+    static const std::vector<ss::sstring> names{
+      cluster::archival_stm_snapshot,
+      cluster::tm_stm_snapshot,
+      cluster::id_allocator_snapshot,
+      cluster::rm_stm_snapshot};
+
+    return names;
+}
+ss::future<> do_copy_persistent_stm_state(
```

Comment:
small nit: if the only thing that we are moving is the stm name list, we can leave `do_copy` and `do_remove` in raft, where they belong more